app/goprotoc/codegen.go

```diff
@@ -38,61 +39,150 @@ var protocOutputs = map[string]struct{}{
 	"ruby":     {},
 }
 
+type outputLocation struct {
+	path         string
+	isZip, isJar bool
```

Comment:
This might be better done as a set of consts:

```go
const (
  fileTypeDir = iota
  fileTypeJar
  fileTypeZip
)
```

An `outputLocation` doesn't make sense with `isZip == true && isJar == true`

Reply:
LOL. You caught me being lazy programmer :P

Reply:
Heh happens to all of us :-)